Remotion/Web/Core/UI/Globalization/ResourceDispatcher.cs

```diff
@@ -132,28 +132,38 @@ public static void Dispatch (Control control, IDictionary elements, string resou
   ///   Dispatches the resources passed in <paramref name="values"/> to the properties of <paramref name="obj"/>.
   /// </summary>
   /// <include file='..\..\doc\include\ResourceDispatcher.xml' path='/ResourceDispatcher/DispatchGeneric/*' />
-  public static void DispatchGeneric (object obj, IDictionary values)
+  public static void DispatchGeneric (object obj, IDictionary<string, WebString> values)
   {
     ArgumentUtility.CheckNotNull ("obj", obj);
     ArgumentUtility.CheckNotNull ("values", values);
 
-    foreach (DictionaryEntry entry in values)
+    foreach (var entry in values)
     {
-      string propertyName = (string) entry.Key;
-      string? propertyValue = (string?) entry.Value;
+      var propertyName = entry.Key;
+      var propertyValue = entry.Value;
 
-      PropertyInfo? property = obj.GetType ().GetProperty (propertyName, typeof (string));
-      if (property != null)
+      PropertyInfo? stringProperty = obj.GetType ().GetProperty (propertyName, typeof (string));
+      PropertyInfo? webStringProperty = obj.GetType ().GetProperty (propertyName, typeof (WebString));
+      PropertyInfo? plainTextStringProperty = obj.GetType ().GetProperty (propertyName, typeof (PlainTextString));
+      if (stringProperty != null)
       {
-        property.SetValue (obj, propertyValue, new object[0]); 
+        stringProperty.SetValue (obj, propertyValue.GetValue(), Array.Empty<object>());
+      }
+      else if (webStringProperty != null)
+      {
+        webStringProperty.SetValue (obj, propertyValue, Array.Empty<object>());
+      }
+      else if (plainTextStringProperty != null)
+      {
+        plainTextStringProperty.SetValue (obj, propertyValue.ToPlainTextString(), Array.Empty<object>());
       }
```

Comment:
Performance consideration: I think it would be best to either use
```
if (obj.GetType ().GetProperty (propertyName, typeof (WebString)) is { } webStringProperty)
{
}
if (obj.GetType ().GetProperty (propertyName, typeof (PlainTexttring)) is { } painTextStringProperty)
{
}
if (obj.GetType ().GetProperty (propertyName, typeof (string)) is { } stringProperty)
{
}
```
(with WebString tested first to optimize the most-likely case)

or, actually better
```
PropertyInfo? property = obj.GetType ().GetProperty (propertyName)
```
and then use IF  (or pattern matching) on the return type for string/webstring/plaintextstring. This way, we only ever check once for the property and get the best possible performance.